examples/third_party/openssl/BUILD.openssl.bazel

```diff
@@ -1,56 +1,178 @@
-"""An openssl build file based on a snippet found in the github issue:
-https://github.com/bazelbuild/rules_foreign_cc/issues/337
-"""
+# Read https://wiki.openssl.org/index.php/Compilation_and_Installation
 
 load("@rules_foreign_cc//foreign_cc:defs.bzl", "configure_make")
 
-# Read https://wiki.openssl.org/index.php/Compilation_and_Installation
+# Note that this dependency name will obviously change if used outside of `rules_foreign_cc`
+load("@rules_foreign_cc_examples_third_party//openssl:openssl_repositories.bzl", "host_transitioned_foreign_cc_target")
 
 filegroup(
     name = "all_srcs",
     srcs = glob(["**"]),
 )
 
+# This wrapper is used to do some last minute configurations to the environment the ForeignCc
+# action runs under for windows. This target should not be used on any other platform.
+genrule(
+    name = "win_tool_prefix",
+    outs = ["win_tool_prefix.sh"],
+    cmd = """\
+cat << EOF > $@
+#!/bin/bash
+set -xeuo pipefail
+
+# The mingw64 target strugles with windows paths so to avoid this
+# the paths are normalized for unix if possible
+function normalize() {
+    type -t cygpath.exe > /dev/null && cygpath.exe "\\$$@" -w || echo "\\$$@"
+}
+
+# Normalize the paths to cerntain executables
+
+if [[ -n "\\$${MAKE:-}" ]]; then
+    export MAKE="\\$$(normalize "\\$${MAKE}")"
+fi
+
+if [[ -n "\\$${AR:-}" ]]; then
+    export AR="\\"\\$$(normalize "\\$${AR}")\\""
+fi
+
+if [[ -n "\\$${CC:-}" ]]; then
+    export CC="\\"\\$$(normalize "\\$${CC}")\\""
+fi
+
+if [[ -n "\\$${CXX:-}" ]]; then
+    export CXX="\\"\\$$(normalize "\\$${CXX}")\\""
+fi
+
+if [[ -n "\\$${NASM:-}" ]]; then
+    export PATH="\\$$(dirname "\\$$NASM");\\$$PATH"
+    export NASM="\\"\\$$(normalize "\\$${NASM}")\\""
+fi
+
+# Normalize the exec root for the action
+export EXT_BUILD_ROOT="\\$$(normalize "\\$${EXT_BUILD_ROOT}")"
+
+# The configure script needs to be executed with perl. If this is the first
+# argument, pass all commands to perl.
+if [[ \\$$1 == *"Configure" ]]; then
+    "\\$${EXT_BUILD_ROOT}/$(execpath @strawberry_perl//:perl)" \\$$@
+
+    # For some reason, the linker arg '/nologo' gets written out as a
+    # path to some file on disk. This corrects for that
+    sed -i "s@C:/tools/msys64/nologo@/nologo@g" \\$$BUILD_TMPDIR/makefile
```

Comment:
I faced this annoying issue also. It’s MSYS2 rewriting `/` to `C:\msys64`. This shouldn’t be happening given that the build script produced by rules_foreign_cc set the `MSYS_NO_PATHCONV` variable - must be a bug with MSYS2. 

To overcome this, I added some code to rules_foreign_cc to rewrite MSVC compiler/linker args so slashes are replaced with dashes, eg `/nologo` with `-nologo`